File: part-4/regression_template.r
# Regression Template

# Libraries
library(ggplot2)

# Importing dataset
dataset = read.csv('')
dataset = dataset[]

# Fitting Regression model to the dataset
# Create your regressor
regressor

# Predicting a new result with Polynomial Regression
y_pred = predict(regressor, data.frame( = ))

# Visualising the Regression model results
ggplot() +
  geom_point(x = , y = , colour = 'red') +
  geom_line(aes(x = , y = predict(regressor, newdata = )),
            colour = 'blue') +
  ggtitle('') +
  xlab('') +
  ylab('')

# Visualising the Regression Model Results (for higher resultions and smoother curve)
x_grid = seq(min(), max(), 0.1)
ggplot() +
  geom_point(x = , y = , colour = 'red') +
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(? = x_grid))),
            colour = 'blue') +
  ggtitle('') +
  xlab('') +
  ylab('')

File: 4_Polynomial_Regression/regression_template.py
# Regression Template

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Importing the dataset
dataset = pd.read_csv('XXXX.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

# Fitting the Regression model to the dataset
# Create your regressor HERE

# Predict a new result with Linear Regression
y_pred = regressor.predict([ 6.5 ])

# Visualising the Regression results
plt.scatter(X, y, color = 'red')
plt.plot(X, regressor.predict(X), color = 'blue')
plot.title('')
plot.xlabel('')
plot.ylabel('')
plt.show()


# Visualising the Regression results (for higher resolution and smoother curve)
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape({len(X_grid, 1)})

plt.scatter(X, y, color = 'red')
plt.plot(X_grid, regressor.predict(X_grid), color = 'blue')
plot.title('')
plot.xlabel('')
plot.ylabel('')
plt.show()